qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/test/test_call_G2sc.py

```diff
@@ -0,0 +1,290 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import unittest
+from unittest import mock
+import os
+import shutil
+
+from mantidqtinterfaces.Engineering.gui.engineering_diffraction.tabs.gsas2.call_G2sc import (
+    add_histograms,
+    add_pawley_reflections,
+    set_max_number_cycles,
+    add_phases,
+    enable_background,
+    enable_histogram_scale_factor,
+    enable_unit_cell,
+    enable_limits,
+    run_microstrain_refinement,
+    run_parameter_refinement,
+    export_refinement_to_csv,
+    export_reflections,
+    export_refined_instrument_parameters,
+    export_lattice_parameters,
+)
+
+import numpy as np
+
+
+class GSAS2ViewTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.temp_save_directory = os.path.join(os.path.dirname(__file__), "temp")
+        try:
+            os.makedirs(cls.temp_save_directory)
+        except FileExistsError:
+            pass
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        shutil.rmtree(cls.temp_save_directory)
+
+    def test_add_phases(self):
+        project = mock.Mock()
```

Comment:
Nitpicking here, but basically every test has this
`project = mock.Mock()`
Perhaps it could go in a `setUp` function that is run before every test?